community/function/src/main/java/org/neo4j/function/Supplier.java

```diff
@@ -24,6 +24,7 @@
  *
  * @param <T> the type of results supplied by this supplier
  */
+@Deprecated // Use java.util.function.Supplier instead
```

Comment:
@jakewins I'd argue that only to be true for `2.3.0`, at least if it is public API.
If it is private, we can do whatever we want with it, whenever we want to.